src/customJs/tools/abandoned_cart_tool/ProductViewer.test.jsx

```diff
@@ -0,0 +1,352 @@
+import { React } from '../../unlayer-react';
+import { render, screen } from '@testing-library/react';
+import { ProductViewer } from './ProductViewer';
+
+describe(ProductViewer.name, () => {
+  it('should render the main container with horizontal layout', async () => {
+    const values = {
+      backgroundColor: 'rgb(255, 255, 255)',
+      layout: '00_horizontal',
+      imageShown: true,
+    };
+
+    render(<ProductViewer values={values} />);
+    const container = await screen.findByRole('container');
+    expect(container.style.backgroundColor).toEqual(values.backgroundColor);
+    expect(container.style.display).toEqual('block');
+    const imageContainer = screen.getByTestId('image-container');
+    expect(imageContainer.style.display).toEqual('inline-block');
+    expect(imageContainer.style.width).toEqual('40%');
+    const descriptionContainer = screen.getByTestId('description-container');
+    expect(descriptionContainer.style.display).toEqual('inline-block');
+    expect(descriptionContainer.style.width).toEqual('55%');
+  });
+
+  it('should render the main container with vertical layout', async () => {
+    const values = {
+      backgroundColor: 'rgb(255, 255, 255)',
+      layout: '01_vertical',
+      imageShown: true,
+    };
+
+    render(<ProductViewer values={values} />);
+    const imageContainer = screen.getByTestId('image-container');
+    expect(imageContainer.style.display).toEqual('block');
+    expect(imageContainer.style.width).toEqual('100%');
+    const descriptionContainer = screen.getByTestId('description-container');
+    expect(descriptionContainer.style.display).toEqual('block');
+    expect(descriptionContainer.style.width).toEqual('100%');
+  });
+
+  describe('image container', () => {
+    it('should render image container with style display none when imageShown is false', async () => {
+      const values = {
+        imageShown: false,
+      };
+
+      render(<ProductViewer values={values} />);
+      const imageContainer = screen.getByTestId('image-container');
+      expect(imageContainer.style.display).toEqual('none');
+    });
+
+    it('should render image container when imageShown is true with full width', async () => {
```

Comment:
Me parece que quedaría mejor asi, al menos para que quede con una estructura similar a la anterior
```suggestion
    it('should render image container with full width when imageShown is true', async () => {
```